### PR TITLE
NCCL should not be dependent on ROCm builds since RCCL is used instead.

### PR DESCRIPTION
**Pull request by rsanthanam-amd**
Allow the nccl flag to be specified even for ROCm builds so that NCCL kernels are included in the build.

In this case the NCCL kernels will use RCCL as the backend implementation.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Would it make sense to add a `+rccl` variant? Or should we use `depends_on("rccl", when="+nccl")`?

**Comment by rsanthanam-amd** [comment]
depends_on("rccl", when="+nccl+rocm") maybe?

not sure if that is the right way to specify that rccl should only be used for rocm builds...

**Comment by adamjstewart** [comment]
Yes, it's already in a `with when("+rocm"):` bracket so that should already be implied.

**Comment by rsanthanam-amd** [comment]
> Would it make sense to add a `+rccl` variant? Or should we use `depends_on("rccl", when="+nccl")`?

Added this. PTAL.

Thanks!

File: var/spack/repos/builtin/packages/py-tensorflow/package.py [hunk 1]
```diff
@@ -345,7 +345,8 @@ class PyTensorflow(Package, CudaPackage, ROCmPackage):
     depends_on("cudnn@:6", when="@0.5:0.6 +cuda")
     depends_on("cudnn@:7", when="@0.7:2.2 +cuda")
     # depends_on('tensorrt', when='+tensorrt')
-    depends_on("nccl", when="+nccl")
+    depends_on("nccl", when="+nccl+cuda")
+    depends_on("rccl" when="+nccl")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py]: I think you'll want to modify the existing `depends_on("rccl")` line, not this one

Comment (rsanthanam-amd) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py]: Makes sense.  Fixed.  Sorry about that.

File: var/spack/repos/builtin/packages/py-tensorflow/package.py [hunk 2]
```diff
@@ -289,7 +289,7 @@ class PyTensorflow(Package, CudaPackage, ROCmPackage):
         depends_on("rocblas")
         depends_on("rocfft")
         depends_on("hipfft")
-        depends_on("rccl")
+        depends_on("rccl" when"+nccl")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py]:
```suggestion
        depends_on("rccl", when"+nccl")
```

Comment (rsanthanam-amd) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py]: Just noticed that.  fixed.

**Review by adamjstewart — DISMISSED** [review]
LGTM, but I'll give @aweits a chance to review

**Comment by rsanthanam-amd** [comment]
@adamjstewart @aweits 

gentle ping.

**Comment by rsanthanam-amd** [comment]
@adamjstewart 

gentle ping

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**